internal/api/agent.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"log"
+	"time"
+
+	"go-yandex-metrics/internal/storage"
+)
+
+type Agent struct {
+	store storage.MemStorage
+	cfg   AgentCfg
+}
+
+func NewAgent(cfg AgentCfg, store *storage.MemStorage) *Agent {
+	agt := &Agent{
+		cfg:   cfg,
+		store: *store,
+	}
+
+	return agt
+}
+
+func (a *Agent) Start() error {
+	tickerSave := time.NewTicker(time.Duration(a.cfg.PollInterval) * time.Second)
+	tickerSend := time.NewTicker(time.Duration(a.cfg.ReportInterval) * time.Second)
+
+	for {
+		select {
+		case <-tickerSave.C:
+			err := a.SaveMetrics()
+			if err != nil {
+				log.Println("failed to save metrics: %w", err)
```

Comment:
```suggestion
				log.Printf("failed to save metrics: %v", err)
```

Обратите внимание, что в коде есть и другие места с этой проблемой. Поправьте их, пожалуйста

Reply:
Прощу прощения - я "сломал" гитхаб у себя на компьютере, и эти изменения пропали (я снова заливал iter5 с гитхаба после "поломки"), и не догадался их проверить перед отправкой.

Reply:
Поправил